caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java

```diff
@@ -155,16 +155,14 @@ public URL toURL(int serialNum, URI uri, Cutout cutout, Map<String, List<String>
             Artifact a = query.performQuery(uri);
 
             if (a == null) {
-                StringBuilder path = new StringBuilder();
-                path.append("400");
-                path.append("|text/plain");
-                path.append("|").append("NotFound: " + uri);
-                String msg = Base64.encodeString(path.toString());
+                return errorURL("NotFound: " + uri);
+            }
 
-                URL serviceURL = reg.getServiceURL(sodaURI, Standards.SODA_SYNC_10, AuthMethod.ANON);
-                URL url = new URL(serviceURL.toExternalForm() + "/" + msg);
-                URL loc = new URL(url.toExternalForm().replace("/sync", "/soda-echo"));
-                return loc;
+            StorageResolver resolver = artifactResolver.getStorageResolver(uri);
+            if (resolver instanceof CutoutGenerator) {
```

Comment:
the same errorURL response should be generated if the resolver is not a CutoutGenerator.

Also, you could keep the reference to the CutoutGenerator as a variable to use below and not have to get it again and check type around line~191. Then the UnsupportedOperation below (in the else) would be removed.

Reply:
Updated.